pep-9999.rst

```diff
@@ -0,0 +1,670 @@
+License field is absent or does not contain a valid License Expression.
+
+Publishing tools MAY infer a License Expression from the provided Classifiers
+entries if they are able to do so unambiguously.
+
+However, no new licensing related classifiers will be added, with anyone
+requesting them being directed to use a License Expression in the License field
+instead. Note that the licensing related Classifiers may be deprecated in a
+future PEP.
+
+
+Mapping legacy Classifiers to new License expressions
+'''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+Publishing tools MAY infer or suggest an equivalent `License Expression` from
+the provided License or Classifiers information if they are able to do so
+unambiguously. For instance, if a package only has this license classifier::
+
+    Classifier: `License :: OSI Approved :: MIT License`
+
+Then the corresponding value for License using a valid license expression to
+suggest would be::
+
+    License: MIT
+
+
+Here are mappings guidelines for the legacy classifiers:
+
+- Classifier `License :: Other/Proprietary License` becomes License: `Proprietary` expression.
+
+- Classifier `License :: Public Domain` becomes License: `Public-Domain` expression.
+
+- The generic and ambiguous Classifiers `License :: OSI Approved`
+  and `License :: DFSG approved` do not have an equivalent license expression.
+
+- The generic and sometimes ambiguous Classifiers `License :: Free For Educational Use`,
+  `License :: Free For Home Use`, `License :: Free for non-commercial use`,
+  `License :: Freely Distributable`, `License :: Free To Use But Restricted`,
+  and `License :: Freeware` are mapped to the generic License: `Proprietary` expression.
+
+The remainder of the `Classifiers` using a `License::` prefix map to a simple
+single license expression using the corresponding SPDX license identifiers.
+
+
+Summary of Differences From PEP 566
+===================================
+
+* Metadata-Version is now 2.2.
+
+* Added one new field: ``License-File``
+
+* Updated the documentation of two fields:  ``License`` and ``Classifiers``
+
+
+License Expression Library Reference implementation
+===================================================
+
+To be succesful, `License Expression` need to be supported in tools.
+
+The `license-expression library`_ provides a Python reference implementation
+library to handle `License Expressions` including parsing, validating and
+formatting `License Expressions` using flexible list of license symbols
+(including SPDX license identifiers and extra identifiers referenced here).
+It is licensed under the Apache-2.0 license and is used in a few projects such
+as the SPDX Python tools, the ScanCode toolkit and the FSFE Reuse.software
+project.
+
+If this PEP is accepted, the Python packaging authority should consider
+adopting the `license-expression library`_ as one of its own tools
+<https://github.com/pypa>_.
+
+
+Surveying how we document licenses today in Python
```

Comment:
This should move into the "Appendix" of the document.

Reply:
Done with https://github.com/pombredanne/spdx-pypi-pep/pull/2/commits/da5ea2708cca19f3dd75f01256e8e37d4bf8d7cc